tests/integration/targets/aci_interface_policy_pim/tasks/main.yml

```diff
@@ -0,0 +1,239 @@
+      designed_router_priority: 4294967296
+      state: present
+    ignore_errors: true
+    register: error_dr_priority_out_of_bounds
+
+  - name: Add a fifth pim interface policy - hello interval out of bounds
+    cisco.aci.aci_interface_policy_pim:
+      <<: *aci_info
+      tenant: ansible_tenant
+      pim_interface: ansible_pim_interface_5
+      hello_interval: 18724287
+      state: present
+    ignore_errors: true
+    register: error_helloIntvl_out_of_bounds
+
+  - name: Add a sixth pim interface policy - join prune interval out of bounds
+    cisco.aci.aci_interface_policy_pim:
+      <<: *aci_info
+      tenant: ansible_tenant
+      pim_interface: ansible_pim_interface_6
+      join_prune_interval: 65521
+      state: present
+    ignore_errors: true
+    register: error_jpIntvl_out_of_bounds
+
+  - name: Asserts for pim interface policy creation tasks
+    ansible.builtin.assert:
+      that:
+        - cm_add_pim_interface is changed
+        - cm_add_pim_interface.previous == []
+        - cm_add_pim_interface.current == []
+        - nm_add_pim_interface is changed
+        - nm_add_pim_interface.current.0.pimIfPol.attributes.name == "ansible_pim_interface_1"
+        - nm_add_pim_interface.current.0.pimIfPol.attributes.authT == "ah-md5"
+        - nm_add_pim_interface.current.0.pimIfPol.attributes.ctrl == "border,passive,strict-rfc-compliant"
+        - nm_add_pim_interface.current.0.pimIfPol.attributes.drDelay == "6"
+        - nm_add_pim_interface.current.0.pimIfPol.attributes.drPrio == "2"
+        - nm_add_pim_interface.current.0.pimIfPol.attributes.helloItvl == "60000"
+        - nm_add_pim_interface.current.0.pimIfPol.attributes.jpInterval == "120"
+        - nm_add_pim_interface.sent.pimIfPol.attributes.authKey == "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
+        - nm_add_pim_interface_2 is changed
+        - nm_add_pim_interface_2.previous == []
+        - nm_add_pim_interface_2.current.0.pimIfPol.attributes.name == "ansible_pim_interface_2"
+        - nm_add_pim_interface_2.current.0.pimIfPol.attributes.authT == "none"
+        - nm_add_pim_interface_2.current.0.pimIfPol.attributes.ctrl == ""
+        - nm_add_pim_interface_2.current.0.pimIfPol.attributes.drDelay == "3"
+        - nm_add_pim_interface_2.current.0.pimIfPol.attributes.drPrio == "1"
+        - nm_add_pim_interface_2.current.0.pimIfPol.attributes.helloItvl == "30000"
+        - nm_add_pim_interface_2.current.0.pimIfPol.attributes.jpInterval == "60"
+        - error_dr_relay_out_of_bounds.msg == "Parameter 'designed_router_delay' is only valid in range between 1 and 65535."
+        - error_dr_priority_out_of_bounds.msg == "Parameter 'designed_router_priority' is only valid in range between 1 and 4294967295."
+        - error_helloIntvl_out_of_bounds.msg == "Parameter 'hello_interval' is only valid in range between 1 and 18724286."
+        - error_jpIntvl_out_of_bounds.msg == "Parameter 'join_prune_interval' is only valid in range between 60 and 65520."
+
```

Comment:
shouls we add tests for changing/update attributes?

Reply:
applies to all modules

Reply:
i agree